api/v1alpha1/instaslice_types.go

```diff
@@ -56,105 +33,161 @@ const (
 }
 
-// InstasliceSpec defines the desired state of Instaslice
 type InstasliceSpec struct {
-	// migGpuUuid represents uuid of the mig device created on the gpu
-	// +required
-	MigGPUUUID map[string]string `json:"migGpuUuid"`
-	// allocations represents allocation details of user workloads
-	// +optional
-	Allocations map[string]AllocationDetails `json:"allocations"`
-	// migplacement represents gpu instance, compute instance with placement for a profile
-	// +required
-	Migplacement []Mig `json:"migplacement"`
-	// cpuonnodeatboot represents total amount of cpu present on the node
-	// +required
-	CpuOnNodeAtBoot int64 `json:"cpuonnodeatboot"`
-	// memoryonnodeatboot represents total amount of memory present on the node
-	// +required
-	MemoryOnNodeAtBoot int64 `json:"memoryonnodeatboot"`
+	// podAllocationRequests specifies the allocation requests per pod
+	// +kubebuilder:validation:optional
+	PodAllocationRequests map[types.UID]AllocationRequest `json:"podAllocationRequests"`
+
+	// emulatorMode specifies whether the Instaslice is running in emulator mode
+	// +kubebuilder:validation:Optional
+	EmulatorMode bool `json:"emulatorMode"`
 }
 
-// InstasliceStatus defines the observed state of Instaslice
 type InstasliceStatus struct {
-	// processed represents state of the instaslice object after daemonset creation
-	// +required
-	Processed bool `json:"processed"`
+	// podAllocationResults specify the allocation results per pod
+	// +kubebuilder:validation:optional
+	PodAllocationResults map[types.UID]AllocationResult `json:"podAllocationResults"`
+
+	// nodeResources specifies the discovered resources of the node
+	// +kubebuilder:validation:optional
+	NodeResources DiscoveredNodeResources `json:"nodeResources"`
+
+	// conditions represent the observed state of the Instaslice
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
 //+kubebuilder:object:root=true
 //+kubebuilder:subresource:status
 
 // Instaslice is the Schema for the instaslices API
+// +kubebuilder:validation:Required
+// +kubebuilder:subresource:status
 type Instaslice struct {
-	// TypeMeta contains metadata about the API resource type.
-	// It includes information such as API version and kind.
-	metav1.TypeMeta `json:",inline"`
-	// +optional
+	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	// spec defines the current state of different allocations in the Instaslice resource.
-	// It contains configuration details, such as GPU allocation settings,
-	// node-specific parameters, and placement preferences.
-	// +required
-	Spec InstasliceSpec `json:"spec"`
-	// status represents the observed state of the Instaslice resource.
-	// It provides runtime information about the resource, such as whether
-	// allocations have been processed.
-	// +optional
-	Status InstasliceStatus `json:"status"`
+
+	Spec   InstasliceSpec   `json:"spec"`
+	Status InstasliceStatus `json:"status,omitempty"`
```

Comment:
The suggestion is to add an optional field named available profile count under spec; this shows possible profile counts available on the node and, for example, `1g.5g:7` and `4g.20gb:1' ... for a single GPU node.